### PR TITLE
feat: add support to properly work with shards with docarray v2

### PR DESCRIPTION
**Pull request by JoanFM**
May depend on https://github.com/docarray/docarray/pull/1437/files

### CODE REVIEW

File: docs/docarray-v2.md [hunk 1]
```diff
@@ -115,12 +113,27 @@ with Deployment(uses=MyExec) as dep:
...
+
+At this moment, these are the features supported if the APIs described in the previous sections are used.
+
+- All the features offered by {ref}`Deployment <deployment>` where a single Executor is served and {ref}`scaled <scale-out>`. This support includes the usage of both
+HTTP and gRPC protocols and the usage of both of them at the same type.
+
+- When combining multiple Deployments in a pipeline using a {ref}`Flow <flow-cookbook>`, multiple limitations exist at this moment:
+
+    - Only gRPC protocol is supported.
+    - Only linear Flows are supported, no topologies using bifurcations can be used at the moment.
```
Comment (samsja) [inline docs/docarray-v2.md:128]: why is that ?

Comment (JoanFM) [reply docs/docarray-v2.md:128]: reduce method is not added for now in the Flow, this will be worked on later

File: docs/docarray-v2.md [hunk 2]
```diff
@@ -115,12 +113,27 @@ with Deployment(uses=MyExec) as dep:
...
+    - There must exist compatibility between the output schema of an Executor and the input schema of the next one.
```
Comment (samsja) [inline docs/docarray-v2.md:129]: IMO this is not a limitation. DocArray v1 did not solve this problem either, v2 just make it explicit. 

We should rather create a note to say that in order for two executor to work in a flow they need to match output/intput schema

Comment (JoanFM) [reply docs/docarray-v2.md:129]: can you please propose how you would change it? I am happy to merge that instead.

**Review by samsja — COMMENTED** [review]
added some comments

File: jina/serve/runtimes/head/request_handling.py [hunk 3]
```diff
@@ -16,6 +16,59 @@
...
+                field_type = bool
+            elif field_type == 'array':
+                field_item_type = field_schema.get('items', {}).get('type', None)
+                if field_item_type == 'string':
+                    field_type = List[str]
+                elif field_item_type == 'integer':
+                    field_type = List[int]
+                elif field_item_type == 'number':
+                    field_type = List[float]
+                elif field_item_type == 'boolean':
```
[the remark this answers is not in the document]
Comment (JoanFM) [reply jina/serve/runtimes/head/request_handling.py:48]: ChatGPT generated this for me and worked

File: docs/docarray-v2.md (hunk 1 above)
Comment (samsja) [reply docs/docarray-v2.md:128]: makes sense

File: docs/docarray-v2.md (hunk 2 above)
Comment (samsja) [inline docs/docarray-v2.md:129]:
```suggestion

````{admonition} Note
:class: note

With DocArray 0.30 support, Jina introduced the concept of input/output schema at the Executor level. In order to chain multiple Executor into a Flow you always need to make sure that the output schema of an Executor is the same as the Input of the Executor that follows him in the Flow
````

**Review by samsja — APPROVED** [review]

File: docs/docarray-v2.md [hunk 4]
```diff
@@ -131,7 +131,14 @@ HTTP and gRPC protocols and the usage of both of them at the same type.
 :class: note
 
 With DocArray 0.30 support, Jina introduced the concept of input/output schema at the Executor level. In order to chain multiple Executor into a Flow you always need to make sure that the output schema of an Executor is the same as the Input of the Executor that follows him in the Flow
+```
 
+````{admonition} Note
+:class: note
+
+For now, Jina Hub will not automatically build your Docker images with the new DocArray version. If this is needed, you need to specifically provide your
```
Comment (alexcg1) [inline docs/docarray-v2.md:139]:
```suggestion
For now, [Executor Hub](https://cloud.jina.ai/executors] will not automatically build your Docker images with the new DocArray version. If this is needed, you need to specifically provide your 
```

File: docs/docarray-v2.md [hunk 5]
```diff
@@ -131,7 +131,14 @@ HTTP and gRPC protocols and the usage of both of them at the same type.
 :class: note
 
 With DocArray 0.30 support, Jina introduced the concept of input/output schema at the Executor level. In order to chain multiple Executor into a Flow you always need to make sure that the output schema of an Executor is the same as the Input of the Executor that follows him in the Flow
+```
 
+````{admonition} Note
+:class: note
+
+For now, Jina Hub will not automatically build your Docker images with the new DocArray version. If this is needed, you need to specifically provide your 
+Dockerfile where the docarray version >=0.30 is specifically installed.
```
Comment (alexcg1) [inline docs/docarray-v2.md:140]:
```suggestion
Dockerfile where `docarray>=0.30` is specifically installed.
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/docarray-v2.md [hunk 6]
```diff
@@ -136,8 +136,8 @@ With DocArray 0.30 support, Jina introduced the concept of input/output schema a
 ````{admonition} Note
 :class: note
 
-For now, Jina Hub will not automatically build your Docker images with the new DocArray version. If this is needed, you need to specifically provide your 
-Dockerfile where the docarray version >=0.30 is specifically installed.
+For now, [Executor Hub](https://cloud.jina.ai/executors] will not automatically build your Docker images with the new DocArray version. If this is needed, you need to specifically provide your
```
Comment (alexcg1) [inline docs/docarray-v2.md:139]:
```suggestion
For now, [Executor Hub](https://cloud.jina.ai/executors] will not automatically build your Docker images with the new DocArray version. If this is needed, you need to provide your 
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/docarray-v2.md [hunk 7]
```diff
@@ -115,12 +113,38 @@ with Deployment(uses=MyExec) as dep:
 ```
 
 
+## Compatible features
+
+Jina is working to offer full compatibility with the new DocArray version.
```
Comment (samsja) [inline docs/docarray-v2.md:118]: I find this sentence a bit weird. the `working` is weird. @alexcg1 wdyt ?

**Pull request merged**